File: frontend/models/cart/CartItemDetailVm.ts
import {ProductOptionValueVm} from "@/models/product/options/ProductOptionValueVm";

export type CartItemDetailVm = {
    productId:number;
    quantity:number;
    productName: string;
    slug: string;
    avatarUrl: string;
    price: number;
    productOptions: ProductOptionValueVm[]
}

File: cart/src/main/java/com/lexuancong/cart/viewmodel/cartitem/CartItemDetailVm.java
package com.lexuancong.cart.viewmodel.cartitem;

import com.lexuancong.cart.viewmodel.productoption.ProductOptionValueVm;
import lombok.Builder;

import java.math.BigDecimal;
import java.util.List;

@Builder
public record CartItemDetailVm(
        Long productId,
        int quantity,
        String  productName,
        String slug,
        String avatarUrl,
        BigDecimal price,
        List<ProductOptionValueVm> productOptionValue

) {
}
